webapp/src/features/adminAddRsu/AdminAddRsu.js

```diff
@@ -0,0 +1,262 @@
+import React, { useEffect } from "react";
+import { Form } from "react-bootstrap";
+import { useForm } from "react-hook-form";
+import { Multiselect, DropdownList } from "react-widgets";
+import {
+  selectSuccessMsg,
+  selectErrorState,
+  selectErrorMsg,
+  selectPrimaryRoutes,
+  selectSelectedRoute,
+  selectOtherRouteDisabled,
+  selectRsuModels,
+  selectSelectedModel,
+  selectSshCredentialGroups,
+  selectSelectedSshGroup,
+  selectSnmpCredentialGroups,
+  selectSelectedSnmpGroup,
+  selectOrganizations,
+  selectSelectedOrganizations,
+  selectSubmitAttempt,
+
+  // actions
+  getRsuCreationData,
+  submitForm,
+  updateSelectedRoute,
+  updateSelectedModel,
+  updateSelectedSshGroup,
+  updateSelectedSnmpGroup,
+  updateSelectedOrganizations,
+} from "./adminAddRsuSlice";
+import { useSelector, useDispatch } from "react-redux";
+
+import "../adminRsuTab/Admin.css";
+
+const AdminAddRsu = (props) => {
+  const dispatch = useDispatch();
+
+  const successMsg = useSelector(selectSuccessMsg);
+  const errorState = useSelector(selectErrorState);
+  const errorMsg = useSelector(selectErrorMsg);
+  const primaryRoutes = useSelector(selectPrimaryRoutes);
+  const selectedRoute = useSelector(selectSelectedRoute);
+  const otherRouteDisabled = useSelector(selectOtherRouteDisabled);
+  const rsuModels = useSelector(selectRsuModels);
+  const selectedModel = useSelector(selectSelectedModel);
+  const sshCredentialGroups = useSelector(selectSshCredentialGroups);
+  const selectedSshGroup = useSelector(selectSelectedSshGroup);
+  const snmpCredentialGroups = useSelector(selectSnmpCredentialGroups);
+  const selectedSnmpGroup = useSelector(selectSelectedSnmpGroup);
+  const organizations = useSelector(selectOrganizations);
+  const selectedOrganizations = useSelector(selectSelectedOrganizations);
+  const submitAttempt = useSelector(selectSubmitAttempt);
+
+  const {
+    register,
+    handleSubmit,
+    reset,
+    formState: { errors },
+  } = useForm();
+
+  useEffect(() => {
+    dispatch(getRsuCreationData());
+  }, []);
```

Comment:
Should we be passing dispatch into the dependency array?

Reply:
added